### ISSUE TITLE
Diffuse color is not loaded from GLB model exported from blender

### ISSUE THREAD

**Issue by Mikalai** [vsg-dev/vsgXchange#118]
When loading GLB model without texture with base color only base color is ignored and model is shown white in vsgviewer. I have assimp 5.2.4 from vcpkg.  

Looks like aiMaterial::Get has several overloads and when called with arbitrary type (like vsg::vec4) this implementation is used
```

material.inl

template <typename Type>
AI_FORCE_INLINE aiReturn aiMaterial::Get(const char* pKey,unsigned int type,
        unsigned int idx,Type& pOut) const {
    const aiMaterialProperty* prop;
    const aiReturn ret = ::aiGetMaterialProperty(this,pKey,type,idx,
        (const aiMaterialProperty**)&prop);
    if ( AI_SUCCESS == ret ) {

        if (prop->mDataLength < sizeof(Type)) {
            return AI_FAILURE;
        }

        if (prop->mType != aiPTI_Buffer) {
            return AI_FAILURE;
        }

        ::memcpy( &pOut, prop->mData, sizeof( Type ) );
    }
    return ret;
}
```

When vsg::vec4 is passed to this function for base color and other color properties condition prop->mType != aiPTI_Buffer is true and data is not copied to vsg::vec4. For base color mType is aiPTI_Float. Though sizeof(vsg::vec4) == prop->mDataLength. This leads to the issue in SceneConverter::convert(const aiMaterial* material, vsg::DescriptorConfig& convertedMaterial) in any place where vsg::vec4 is passed for the property which has mType equals to aiPTI_Float. Type cast of vsg::vec4 to aiColor4D fixed the issue as another overload of aiMaterial::Get is used in this case, though not sure if it is the best way to fix this.

After this changes in assimp.cpp color was loaded and depicted correctly, same as in blender.
```

@@ -288,22 +288,22 @@ void SceneConverter::convert(const aiMaterial* material, vsg::DescriptorConfig&
     if ((options->getValue(assimp::two_sided, convertedMaterial.two_sided) || (material->Get(AI_MATKEY_TWOSIDED, convertedMaterial.two_sided) == AI_SUCCESS)) && convertedMaterial.two_sided)
     {
         defines.push_back("VSG_TWO_SIDED_LIGHTING");
     }
 
-    if (material->Get(AI_MATKEY_BASE_COLOR, pbr.baseColorFactor) == AI_SUCCESS || hasPbrSpecularGlossiness)
+    if (material->Get(AI_MATKEY_BASE_COLOR, *(aiColor4D*)(& pbr.baseColorFactor)) == AI_SUCCESS || hasPbrSpecularGlossiness)
     {
         // PBR path
         convertedMaterial.shaderSet = getOrCreatePhrShaderSet();
 
         if (convertedMaterial.blending)
             pbr.alphaMask = 0.0f;
 
         if (hasPbrSpecularGlossiness)
         {
             defines.push_back("VSG_WORKFLOW_SPECGLOSS");
-            material->Get(AI_MATKEY_COLOR_DIFFUSE, pbr.diffuseFactor);
+            material->Get(AI_MATKEY_COLOR_DIFFUSE, *(aiColor4D*)(& pbr.diffuseFactor));
 
             if (material->Get(AI_MATKEY_GLOSSINESS_FACTOR, pbr.specularFactor.a) != AI_SUCCESS)
             {
                 if (float shininess; material->Get(AI_MATKEY_SHININESS, shininess))
                     pbr.specularFactor.a = shininess / 1000;
@@ -361,13 +361,13 @@ void SceneConverter::convert(const aiMaterial* material, vsg::DescriptorConfig&
         if (convertedMaterial.blending)
             mat.alphaMask = 0.0f;
 
         material->Get(AI_MATKEY_GLTF_ALPHACUTOFF, mat.alphaMaskCutoff);
         material->Get(AI_MATKEY_COLOR_AMBIENT, mat.ambient);
-        const auto diffuseResult = material->Get(AI_MATKEY_COLOR_DIFFUSE, mat.diffuse);
-        const auto emissiveResult = material->Get(AI_MATKEY_COLOR_EMISSIVE, mat.emissive);
-        const auto specularResult = material->Get(AI_MATKEY_COLOR_SPECULAR, mat.specular);
+        const auto diffuseResult = material->Get(AI_MATKEY_COLOR_DIFFUSE, *(aiColor4D*)(&mat.diffuse));
+        const auto emissiveResult = material->Get(AI_MATKEY_COLOR_EMISSIVE, *(aiColor4D*)(&mat.emissive));
+        const auto specularResult = material->Get(AI_MATKEY_COLOR_SPECULAR, *(aiColor4D*)(&mat.specular));
 
         aiShadingMode shadingModel = aiShadingMode_Phong;
         material->Get(AI_MATKEY_SHADING_MODEL, shadingModel);
 
         unsigned int maxValue = 1;

```
Before changes
![image](https://user-images.githubusercontent.com/410261/196003773-aaa26845-a5ed-4d30-9750-755dc70c1c40.png)
After changes
![image](https://user-images.githubusercontent.com/410261/196003685-af250e90-3d02-4128-bb3c-ab8ab51e2f78.png)

Test model is attached.

Thanks
[solid_color.zip](https://github.com/vsg-dev/vsgXchange/files/9792896/solid_color.zip)

**Comment by robertosfield** [comment]
The suggest cast threw up compile errors on my system but it gave me the hints of what to fix.  I've created a getColor(..) helper function to map aiColor4D to vsg::vec4 and checked this in to a [AssimpDiffuseColor](https://github.com/vsg-dev/vsgXchange/tree/AssimpDiffuseColor) branch for these mods.  

I've tested the changes out on the glTF-Sample-Models set and and your solid_color.glb and it looks better all round to me, but I don't have a reference viewer to compare them against.

The changes are in commit: 17350d63574869a02786d08cb62d00440381822c

Could you test this branch out and if it looks fine I'll merge with vsgXchange master.

**Comment by Mikalai** [comment]
[AssimpDiffuseColor](https://github.com/vsg-dev/vsgXchange/tree/AssimpDiffuseColor) looks good. Model is depicted with diffuse color. 
![image](https://user-images.githubusercontent.com/410261/196492500-48c872ac-9426-402f-be89-f6b1b4c28a78.png)
Similar to 3D Viewer results
![image](https://user-images.githubusercontent.com/410261/196492648-ff0e9863-80e4-400a-8598-5513a8672b1a.png)

**Comment by robertosfield** [comment]
Thanks for the testing, I've merged #119 with vsgXchange master.
